flocker/node/_config.py

```diff
@@ -1190,9 +1190,12 @@ def _parse(self):
             self._applications[application_name] = Application(**attributes)
 
 
-def deployment_from_configuration(deployment_configuration, all_applications):
+def _apply_deployment_to_applications(
```

Comment:
I suspect something like `Cluster.from_deployment_and_application_config` might make sense as the API for this functionality?  (No, `Cluster` does not exist yet.)